src/mytardis_client/common_types.py

```diff
@@ -1,7 +1,12 @@
-# pylint: disable=missing-class-docstring,fixme
-"""Useful enunmerators for ingestion"""
+"""Definitions of fundamental data types used when interacting with MyTardis.
+
+Note that the specifications of the MyTardis objects are not included here; see
+objects.py for that."""
 
 from enum import Enum
+from typing import Literal
+
+HttpRequestMethod = Literal["GET", "POST", "PUT", "DELETE", "PATCH"]
```

Comment:
Just a thought. Do we want to limit this to the actions that we have api calls for? I know we are planning on setting up the PUT/PATCH calls but am not sure we have need for a DELETE call.

Reply:
Good question, I had originally intended this to be a general collection of HTTP methods, but if it's defined in `mytardis_client`, it does make sense for it to be specifically limited to what MyTardis actually supports.

My original intent with adding this `Literal` was to have a stronger interface - callers can't pass an arbitrary string, just an HTTP method name. So it makes sense to constrain it further. I will make that change